packages/runtime/src/interpreter/opcode.ts

```diff
@@ -28,6 +28,17 @@ export class Op {
     }
   }
 
+  /**
+   * asserts number is less than or equal to MAX_UINT128
+   * @param num number to check
+   * @param line line number in TEAL file
+   */
+  checkOverflowU128 (num: bigint, line: number): void {
+    if (num > MAX_UINT128) {
+      throw new RuntimeError(RUNTIME_ERRORS.TEAL.UINT128_OVERFLOW, { line: line });
+    }
+  }
```

Comment:
Why not use `checkOverflow` function above? We could generalize  it as `checkUintOverflow` & pass `MAX_UINT64`/`MAX_UINT128`. Errors could be generalized as well.

Reply:
ok